src/SpacetimeDBClient.cs

```diff
@@ -146,29 +145,38 @@ struct PreProcessedMessage
         private readonly CancellationTokenSource _preProcessCancellationTokenSource = new();
         private CancellationToken _preProcessCancellationToken => _preProcessCancellationTokenSource.Token;
 
+        static DbValue Decode(ClientCache.ITableCache table, EncodedValue value) => value switch
+        {
+            EncodedValue.Binary(var bin) => new DbValue(table.DecodeValue(bin), bin),
+            EncodedValue.Text(var text) => throw new InvalidOperationException("JavaScript messages aren't supported."),
+            _ => throw new InvalidOperationException(),
+        };
+
         void PreProcessMessages()
         {
-            while (!isClosing)
+            try
             {
-                try
```

Comment:
Why did we swap the ordering of this? It was intentional that we had the try on the inside.